src/main/java/hudson/plugins/git/util/Build.java

```diff
@@ -92,6 +92,21 @@ public Result getBuildResult() {
     }
 
     @Override
+    public boolean equals(Object o) {
+        if (!(o instanceof Build)) {
+            return false;
+        } else {
+            Build otherBuild = (Build) o;
+            if (otherBuild.hudsonBuildNumber == this.hudsonBuildNumber
+                    && otherBuild.revision.equals(this.revision)) {
+                return true;
```

Comment:
Pushed a change with `marked` added, and moved to `==` rather than `.equals` to deal with null case.